### PR TITLE
Disable stake button when delegate does not accept staking

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
- Moved the bakerList up the tree to accountBody 
- added delegate address in the DelegateData.
- added logic in Stake button.

- [x] Disable stake button if the unstake requests from another baker are not finalizable.

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ export interface DelegateData {
   spendableBalance: number
   stakedBalance: number
   unstakedBalance: number
+  delegate: string | null
```
Comment (rickykuo666) [inline pages/index.tsx:20]: 1. can we rename this to `bakerAddress`? for better naming
2. i think it shoule be bakerAddress?: string

File: pages/index.tsx [hunk 2]
```diff
@@ -54,7 +55,8 @@ export default function Home() {
           data.balance - data.stakedBalance - data.unstakedBalance
         ),
         stakedBalance: mutezToTez(data.stakedBalance),
-        unstakedBalance: mutezToTez(data.unstakedBalance)
+        unstakedBalance: mutezToTez(data.unstakedBalance),
+        delegate: data.delegate.address
```
Comment (rickykuo666) [inline pages/index.tsx:59]: if not delegate, will have runtime error. you can use data.delegate?.address

**Review by rickykuo666 — CHANGES_REQUESTED** [review]
in `ChangeBaker/index.tsx` there is one extra getBakerList query needs to remove

File: pages/index.tsx (hunk 1 above)
Comment (ajinkyaraj-23) [reply pages/index.tsx]: Removed it as we did not need it.

File: pages/index.tsx (hunk 2 above)
Comment (ajinkyaraj-23) [reply pages/index.tsx:58]: Removed it as we dont need it.

File: components/Operations/tezConnections.ts [hunk 3]
```diff
@@ -106,10 +104,20 @@ export function updateStakingOpsStatus(
           totalFinalizableAmount += operation.remainingFinalizableAmount
         }
       } else {
+        if (!prevBakerAddress) {
```
Comment (rickykuo666) [inline components/Operations/tezConnections.ts:107]: I think `prevBakerAddress` is always gonna be null, isn't it? no need for this check

File: components/AccountBody.tsx [hunk 4]
```diff
@@ -212,7 +244,7 @@ export const AccountBody = ({
               <Text fontSize='14px' color='#4A5568' fontWeight={600}>
                 BAKER
               </Text>
-              {!!stakingOpsStatus.Delegated ? (
+              {Boolean(stakingOpsStatus.Delegated) ? (
```
Comment (rickykuo666) [inline components/AccountBody.tsx:247]: we can get rid of Boolean or !!, cuz we 're checking it's true

**Review by rickykuo666 — CHANGES_REQUESTED** [review]
`/ChangeBaker/index.tsx` still has an exrac getBakerlist

**Review by rickykuo666 — APPROVED** [review]
LGTM

**Pull request merged**